### PR TITLE
Add TryCreate to Identifier structures

### LINKED ISSUE #23146: [FEATURE REQ] KeyVaultKeyIdentifier.TryParse
**Library or service name.**
Azure.Security.KeyVault.Keys
 
**Is your feature request related to a problem? Please describe.**
If I develop an application that takes user input that should be a key vault key URI I have to write the following code to validate it is a Key Vault URI.

```
try
{
    identifier = new KeyVaultKeyIdentifier(id);
}
catch (ArgumentException)
{
    // Invalid id
}
```

Ideally I would like to write

```
if (KeyVaultKeyIdentifier.TryParse(id, out KeyVaultKeyIdentifier identifier))
{
    // use it
}
else
{
    // Return error
}
```

### PR DESCRIPTION
**Pull request by heaths**
Fixes #23146

### CODE REVIEW

File: eng/scripts/Export-API.ps1 [hunk 1]
```diff
@@ -21,6 +21,7 @@ if ($SpellCheckPublicApiSurface) {
     npx cspell lint `
         --config "$PSScriptRoot/../../.vscode/cspell.json" `
         --no-must-find-files `
+        --root "$PSScriptRoot/../../" `
```
Comment (maririos) [inline eng/scripts/Export-API.ps1:24]: FYI @danieljurek

Comment (heaths) [reply eng/scripts/Export-API.ps1:24]: Without this, the root is the current directory, so if you run this in, say, your service directory (common) the `"overrides"` section of _.github/cspell.json_ won't match the file patterns.

**Review by weshaggard — APPROVED** [review]

**Review by tg-msft — APPROVED** [review]
Looks good!

**Review by KrzysztofCwalina — APPROVED** [review]

File: eng/scripts/Export-API.ps1 (hunk 1 above)
Comment (danieljurek) [reply eng/scripts/Export-API.ps1:24]: Thanks! I've been running all of these from the root of the repo.

**Pull request merged**